### ISSUE TITLE
Oddities with hair color and UI + UE

### ISSUE THREAD

**Issue by dylanstrategie** [vgstation-coders/vgstation13#5054]
I currently use the following color for hair + beard

![example](https://cloud.githubusercontent.com/assets/6137403/8258429/05ec96b0-16b5-11e5-958d-bbba2761a02d.png)

For some reason, whenever I get cloned or have my UI + UE taken and injected into anyone, I end up with purple hair. It's been that way for almost forever

**Comment by Exxion** [comment]
And I once had a character with pure blue hair (yes, #0000FF) whose hair became a much more muted blue after cloning
I don't remember the exact shade, but the difference was blatantly obvious
It was actually a pretty nice color, but very different from what it had been

**Comment by NigglyWiggly** [comment]
That exact same thing happens to me as well
I use the same color, and it's A L W A Y S purple in my UI+UE

**Comment by PJB3005** [comment]
Something might be up with the way colours get converted to UIs

**Comment by clusterfack** [comment]
D3athrow recently created a runtime in UIs, however if this is an old bug it may not be related

**Comment by 9600bauds** [comment]
Definitely a bug and often seen with cloned people

**Comment by ihadtoregisterforthis** [comment]
Seen a green haired clone earlier today. So, it's not only purple.

**Comment by carshalash** [comment]
Wait, I thought this shit was intentional?

**Comment by PJB3005** [comment]
Wasn't this because of something ripping out the # in the colour, causing BYOND to see it as a different colour?

**Comment by Exxion** [comment]
This affects eye color as well, if I'm not mistaken

**Comment by ghost** [comment]
Well, if you know what the problem is, you're invited to fix it, because it's been plaguing this codebase forever and no-one knew how to fix it for the longest time

**Comment by Megamaw** [comment]
FIX IT

**Comment by NigglyWiggly** [comment]
Nah don't bother

**Comment by 9600bauds** [comment]
I'll look at this later I guess

**Comment by 9600bauds** [comment]
How much would people care if I made cloning still give you purple hair so scanning your body in a genetics scanner actually has a purpose

**Comment by Megamaw** [comment]
would players? yes. would geneticists care enough to do it? probably not.

**Comment by NigglyWiggly** [comment]
That'd be neat I guess.
I actually like the whole random hair color thing because it seems like a perfect little quirk for the cloner to have.
Like it's not quite perfect, y'know?

**Comment by Exxion** [comment]
Yeah, but it should at least be intentional
Also I think it's not really random, but always the same color for a given hair color

**Comment by Megamaw** [comment]
do black haired people turn blonde? if not, fix that.

**Comment by ghost** [comment]
Whatever it is, fix the bug that gives you incorrect color. I've even seen it bust a ling once because he was running around as me but with purple fucking hair and some people noticed

It's obviously an unintended bug, maybe a BYOND failure due to a missing # and thus an erroneous interpretation of the value

**Comment by gbasood** [comment]
well then according to @dylanstrategie and others, definitely a bug

**Comment by Megamaw** [comment]
i want genetics to turn me into a Super Saiyan.

**Comment by 9600bauds** [comment]
Tried killing and cloning a dude 6 times, no hair color change. Steps to reproduce would be appreciated

**Comment by NigglyWiggly** [comment]
Literally just clone.
It only happens on certain hair colors, specifically the one he showed.
It's not really too much of an issue though, regardless

**Comment by 9600bauds** [comment]
I tried specifically that hair color, nada. Weird fucking bug

**Comment by NigglyWiggly** [comment]
What?
Maybe it got fixed already, then?

**Comment by 9600bauds** [comment]
The only bug more elusive than the magic walk-restoring punch.

**Comment by Shadowmech88** [comment]
It appears as though #19881 did not fix this.
![image](https://user-images.githubusercontent.com/3991927/67612099-51f0bd80-f765-11e9-9ebe-9a08a4701931.png)
It doesn't seem to happen with every hair color, or at least happens to varying degrees depending on what the color is.
The body I spawned in with had light yellow hair (bowl cut,) which remained the same color both after cloning (directly below, in the briefs,) and after injecting UI+UE into a monkeyman (bottom right.)
However, the body of the guest account I joined with to test the cloning had a dark yellow afro and mustache, which became a lighter yellow afro and white mustache both after cloning and injecting the UI+UE into a monkeyman.
